### PR TITLE
Update to nrfx 3.7.0, add nRF54H20 EngB SoC and nRF54H20 DK 0.9.0

### PR DESCRIPTION
**Pull request by gmarull**
This PR adds nRF54H20 EngB SoC revision (current one) and updates nRF54H20 DK to have two revisions: 0.8.0 (current) and 0.9.0 (new and default, with the new SoC revision).

### CODE REVIEW

File: scripts/ci/check_compliance.py [hunk 1]
```diff
@@ -927,6 +927,7 @@ def check_no_undef_outside_kconfig(self, kconf):
                               # Zephyr toolchain variant and therefore not
                               # visible to compliance.
         "BOARD_", # Used as regex in scripts/utils/board_v1_to_v2.py
+        "BOARD_REVISION", # Used in some Kconfig.board
```
Comment (gmarull) [inline scripts/ci/check_compliance.py:930]: @nordicjm @tejlmand any idea on why this is required? see https://github.com/zephyrproject-rtos/zephyr/actions/runs/10919646498/job/30307751818

Comment (nordicjm) [reply scripts/ci/check_compliance.py:930]: Would assume the hierarchy of how the script runs, it probably just sources board files and not zephyr's Kconfig files

Comment (gmarull) [reply scripts/ci/check_compliance.py]: Actually, this was not a fix, but https://github.com/zephyrproject-rtos/zephyr/pull/78631/commits/fe26d9e4752013681703ee50f688d8450ea87c6a (not sure if the best way, but seems to fix the problem).

**Comment by nika-nordic** [comment]
@gmarull maybe we can switch to use: https://github.com/zephyrproject-rtos/hal_nordic/pull/223 and take this PR out of draft ?

**Comment by gmarull** [comment]
@aescolar  can you check babblesim failures?

File: modules/hal_nordic/nrfx/nrfx_config.h [hunk 2]
```diff
@@ -1042,6 +1039,14 @@
...
 #elif defined(NRF5340_XXAA_NETWORK)
     #include <nrfx_config_nrf5340_network.h>
+#elif defined(NRF54H20_ENGB_XXAA) && defined(NRF_APPLICATION)
+    #include <nrfx_config_nrf54h20_engb_application.h>
+#elif defined(NRF54H20_ENGB_XXAA) && defined(NRF_RADIOCORE)
+    #include <nrfx_config_nrf54h20_engb_radiocore.h>
+#elif defined(NRF54H20_ENGB_XXAA) && defined(NRF_PPR)
+    #include <nrfx_config_nrf54h20_engb_ppr.h>
+#elif defined(NRF54H20_ENGB_XXAA) && defined(NRF_FLPR)
+    #include <nrfx_config_nrf54h20_engb_flpr.h>
```
Comment (anangl) [inline modules/hal_nordic/nrfx/nrfx_config.h:1049]: Do we actually need a second copy of these config files? Why not just use:
```
#elif (defined(NRF54H20_XXAA) || defined(NRF54H20_ENGB_XXAA)) && defined(NRF_APPLICATION)
    #include <nrfx_config_nrf54h20_application.h>
```
and so on?

File: modules/hal_nordic/nrfx/nrfx_config_nrf54h20_application.h [hunk 3]
```diff
@@ -312,7 +313,7 @@
  * Integer value. Minimum: 0. Maximum: 15.
  */
 #ifndef NRFX_GPIOTE_CONFIG_NUM_OF_EVT_HANDLERS
-#define NRFX_GPIOTE_CONFIG_NUM_OF_EVT_HANDLERS 1
+#define NRFX_GPIOTE_CONFIG_NUM_OF_EVT_HANDLERS 2
```
Comment (anangl) [inline modules/hal_nordic/nrfx/nrfx_config_nrf54h20_application.h:316]: This is unwanted change. Similarly for other cores.

File: modules/hal_nordic/nrfx/nrfx_config_nrf54h20_radiocore.h [hunk 4]
```diff
@@ -11,6 +11,10 @@
 #error "This file should not be included directly. Include nrfx_config.h instead."
 #endif
 
+#ifndef NRFX_RTC0_ENABLED
+#define NRFX_RTC0_ENABLED 1
+#endif
+
```
Comment (anangl) [inline modules/hal_nordic/nrfx/nrfx_config_nrf54h20_radiocore.h:17]: This is also unwanted. It should be fixed in nrfx (@nika-nordic).

**Review by aescolar — COMMENTED** [review]
```diff
diff --git a/subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_nrf54lx.h b/subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_nrf54lx.h
index b0d3948aefb..8597ea7d586 100644
--- a/subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_nrf54lx.h
+++ b/subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_nrf54lx.h
@@ -633,9 +633,16 @@ static inline uint32_t hal_radio_tx_power_value(int8_t tx_power_lvl)
                return RADIO_TXPOWER_TXPOWER_Neg20dBm;
        }
 
+#if defined(RADIO_TXPOWER_TXPOWER_Neg26dBm)
        if (tx_power_lvl >= -26) {
                return RADIO_TXPOWER_TXPOWER_Neg26dBm;
        }
+#endif
+#if defined(RADIO_TXPOWER_TXPOWER_Neg28dBm)
+       if (tx_power_lvl >= -28) {
+               return RADIO_TXPOWER_TXPOWER_Neg28dBm;
+       }
+#endif
 
        if (tx_power_lvl >= -40) {
                return RADIO_TXPOWER_TXPOWER_Neg40dBm;
```

**Comment by aescolar** [comment]
And
```diff
diff --git a/modules/hal_nordic/nrfx/CMakeLists.txt b/modules/hal_nordic/nrfx/CMakeLists.txt
index fde999b556a..595da6df111 100644
--- a/modules/hal_nordic/nrfx/CMakeLists.txt
+++ b/modules/hal_nordic/nrfx/CMakeLists.txt
@@ -103,7 +103,7 @@ zephyr_library_sources_ifdef(CONFIG_SOC_NRF52840       ${MDK_DIR}/system_nrf5284
zephyr_library_sources_ifdef(CONFIG_SOC_NRF5340_CPUAPP ${MDK_DIR}/system_nrf5340_application.c)
zephyr_library_sources_ifdef(CONFIG_SOC_NRF5340_CPUNET ${MDK_DIR}/system_nrf5340_network.c)
zephyr_library_sources_ifdef(CONFIG_SOC_SERIES_NRF54HX ${MDK_DIR}/system_nrf54h.c)
-zephyr_library_sources_ifdef(CONFIG_SOC_SERIES_NRF54LX ${MDK_DIR}/system_nrf54l.c)
+zephyr_library_sources_ifdef(CONFIG_SOC_COMPATIBLE_NRF54LX ${MDK_DIR}/system_nrf54l.c)
zephyr_library_sources_ifdef(CONFIG_SOC_SERIES_NRF91X  ${MDK_DIR}/system_nrf91.c)
zephyr_library_sources_ifdef(CONFIG_SOC_SERIES_NRF92X  ${MDK_DIR}/system_nrf92.c)
```

File: modules/hal_nordic/nrfx/CMakeLists.txt [hunk 5]
```diff
@@ -95,7 +103,9 @@ zephyr_library_sources_ifdef(CONFIG_SOC_NRF52840       ${MDK_DIR}/system_nrf5284
 zephyr_library_sources_ifdef(CONFIG_SOC_NRF5340_CPUAPP ${MDK_DIR}/system_nrf5340_application.c)
 zephyr_library_sources_ifdef(CONFIG_SOC_NRF5340_CPUNET ${MDK_DIR}/system_nrf5340_network.c)
 zephyr_library_sources_ifdef(CONFIG_SOC_SERIES_NRF54HX ${MDK_DIR}/system_nrf54h.c)
-zephyr_library_sources_ifdef(CONFIG_SOC_SERIES_NRF54LX ${MDK_DIR}/system_nrf54l.c)
+if(CONFIG_SOC_SERIES_NRF54LX OR CONFIG_SOC_SERIES_BSIM_NRF54LX)
```
Comment (aescolar) [inline modules/hal_nordic/nrfx/CMakeLists.txt:106]: CONFIG_SOC_COMPATIBLE_NRF54LX means that :)

Comment (gmarull) [reply modules/hal_nordic/nrfx/CMakeLists.txt:106]: I think we should avoid intermediate symbols here.

Comment (aescolar) [reply modules/hal_nordic/nrfx/CMakeLists.txt:106]: We are already using those COMPATIBLE ones in many places. They were introduced for this kind of cases.

File: modules/hal_nordic/nrfx/nrfx_config.h (hunk 2 above)
Comment (nika-nordic) [reply modules/hal_nordic/nrfx/nrfx_config.h:1052]: @gmarull yes, nrfx_config_* for these two targets are identical

Comment (gmarull) [reply modules/hal_nordic/nrfx/nrfx_config.h]: adjusted

File: drivers/timer/nrf_grtc_timer.c [hunk 6]
```diff
@@ -513,6 +507,9 @@ static int sys_clock_driver_init(void)
 	nrfx_grtc_clock_source_set(NRF_GRTC_CLKSEL_LFXO);
 #endif
 
+#if IS_ENABLED(CONFIG_NRF_GRTC_ALWAYS_ON)
```
Comment (anangl) [inline drivers/timer/nrf_grtc_timer.c:510]:
```suggestion
#if defined(CONFIG_NRF_GRTC_ALWAYS_ON)
```

File: drivers/timer/Kconfig.nrf_grtc [hunk 7]
```diff
@@ -15,11 +15,10 @@ menuconfig NRF_GRTC_TIMER
 
 if NRF_GRTC_TIMER
 
-config NRF_GRTC_SLEEP_ALLOWED
-	def_bool y
-	depends on POWEROFF
+config NRF_GRTC_ALWAYS_ON
+	def_bool n
```
Comment (anangl) [inline drivers/timer/Kconfig.nrf_grtc:19]: Why not just `bool`? `n` is anyway the default value.

File: drivers/adc/adc_nrfx_saadc.c [hunk 8]
```diff
@@ -19,7 +19,7 @@ LOG_MODULE_REGISTER(adc_nrfx_saadc);
 
 #if (NRF_SAADC_HAS_AIN_AS_PIN)
 
-#if defined(CONFIG_SOC_NRF54H20) || defined(CONFIG_SOC_NRF9280)
+#if defined(CONFIG_SOC_NRF54H20) || defined(CONFIG_SOC_NRF54H20_ENGB) || defined(CONFIG_SOC_NRF9280)
```
Comment (anangl) [inline drivers/adc/adc_nrfx_saadc.c:22]:
```suggestion
#if defined(CONFIG_NRF_PLATFORM_HALTIUM)
```

**Review by nika-nordic — DISMISSED** [review]

**Review by anangl — DISMISSED** [review]

**Review by fabiobaltieri — DISMISSED** [review]

**Review by nika-nordic — DISMISSED** [review]

**Review by tmon-nordic — DISMISSED** [review]

**Review by jaz1-nordic — DISMISSED** [review]

**Comment by nika-nordic** [comment]
https://github.com/zephyrproject-rtos/hal_nordic/pull/223 merged, please update SHA and rebase

**Review by anangl — APPROVED** [review]

**Review by tmon-nordic — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Review by kl-cruz — APPROVED** [review]

**Review by nika-nordic — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Pull request merged**